FFmpegInterop/UncompressedAudioSampleProvider.cpp

```diff
@@ -17,161 +17,152 @@
 //*****************************************************************************
 
 #include "pch.h"
-
 #include "UncompressedAudioSampleProvider.h"
 
-using namespace FFmpegInterop;
-
-// Minimum duration for uncompressed audio samples (200 ms)
-const LONGLONG MINAUDIOSAMPLEDURATION = 2000000;
-
-UncompressedAudioSampleProvider::UncompressedAudioSampleProvider(
-	FFmpegReader^ reader,
-	AVFormatContext* avFormatCtx,
-	AVCodecContext* avCodecCtx)
-	: UncompressedSampleProvider(reader, avFormatCtx, avCodecCtx)
-	, m_pSwrCtx(nullptr)
-{
-}
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Media::MediaProperties;
+using namespace winrt::Windows::Storage::Streams;
+using namespace std;
 
-HRESULT UncompressedAudioSampleProvider::AllocateResources()
+namespace winrt::FFmpegInterop::implementation
 {
-	HRESULT hr = S_OK;
-	hr = UncompressedSampleProvider::AllocateResources();
-	if (SUCCEEDED(hr))
+	UncompressedAudioSampleProvider::UncompressedAudioSampleProvider(_In_ AVStream* stream, _In_ Reader& reader) :
+		UncompressedSampleProvider(stream, reader),
+		m_minAudioSampleDur(ConvertToAVTime(MIN_AUDIO_SAMPLE_DUR_MS, MS_PER_SEC, m_stream->time_base))
 	{
-		// Set default channel layout when the value is unknown (0)
-		int64 inChannelLayout = m_pAvCodecCtx->channel_layout ? m_pAvCodecCtx->channel_layout : av_get_default_channel_layout(m_pAvCodecCtx->channels);
-		int64 outChannelLayout = av_get_default_channel_layout(m_pAvCodecCtx->channels);
-
-		// Set up resampler to convert any PCM format (e.g. AV_SAMPLE_FMT_FLTP) to AV_SAMPLE_FMT_S16 PCM format that is expected by Media Element.
-		// Additional logic can be added to avoid resampling PCM data that is already in AV_SAMPLE_FMT_S16_PCM.
-		m_pSwrCtx = swr_alloc_set_opts(
-			NULL,
-			outChannelLayout,
-			AV_SAMPLE_FMT_S16,
-			m_pAvCodecCtx->sample_rate,
-			inChannelLayout,
-			m_pAvCodecCtx->sample_fmt,
-			m_pAvCodecCtx->sample_rate,
-			0,
-			NULL);
-
-		if (!m_pSwrCtx)
+		if (m_codecContext->sample_fmt != AV_SAMPLE_FMT_S16)
```

Comment:
>m_codecContext->sample_fmt != AV_SAMPLE_FMT_S16 [](start = 6, length = 47)

Similar comment to the uncompressed video sample provider - we support more uncompressed audio formats in the MF pipeline.

Reply:
I filed a task.